test/files/neg/t10507/Macros_1.scala

```diff
@@ -0,0 +1,13 @@
+import scala.language.experimental.macros
+import scala.reflect.macros.blackbox
+
+object Macros_1 {
+  def seq: List[Int] = macro impl
+  def impl(c: blackbox.Context): c.Tree = {
+    import c.universe._
+    q"""{
+      class A(val i: Int) { def this() = this(0: Short); def this(s: Short) = this(s) }
```

Comment:
The Short version is recursive. Is that intentional? I guess we don't see an error because it's the same position as the previous error we see. `s.toInt`.

Reply:
Good catch 👍